src/background/adblocker.js

```diff
@@ -124,41 +125,46 @@ async function updateEngines() {
 
 const HOUR_IN_MS = 60 * 60 * 1000;
 export const setup = asyncSetup([
-  observe(async (value, lastValue) => {
-    options = value;
-
-    const enabledEngines = getEnabledEngines(value);
-    const prevEnabledEngines = lastValue && getEnabledEngines(lastValue);
-
-    if (
-      // Reload/mismatched main engine
-      !(await engines.init(engines.MAIN_ENGINE)) ||
-      // Enabled engines changed
-      (prevEnabledEngines &&
-        (enabledEngines.length !== prevEnabledEngines.length ||
-          enabledEngines.some((id, i) => id !== prevEnabledEngines[i])))
-    ) {
-      // The regional filters engine is no longer used, so we must remove it
-      // from the storage. We do it as rarely as possible, to avoid unnecessary loads.
-      // TODO: this can be removed in the future release when most of the users will have
-      // the new version of the extension
-      engines.remove('regional-filters');
-
-      await reloadMainEngine();
-    }
-
-    if (options.filtersUpdatedAt < Date.now() - HOUR_IN_MS) {
-      await updateEngines();
-    }
-  }),
-  observe('experimentalFilters', async (value, lastValue) => {
-    engines.setEnv('env_experimental', value);
+  OptionsObserver.addListener(
+    async function adblockerEngines(value, lastValue) {
+      options = value;
+
+      const enabledEngines = getEnabledEngines(value);
+      const prevEnabledEngines = lastValue && getEnabledEngines(lastValue);
+
+      if (
+        // Reload/mismatched main engine
+        !(await engines.init(engines.MAIN_ENGINE)) ||
+        // Enabled engines changed
+        (prevEnabledEngines &&
+          (enabledEngines.length !== prevEnabledEngines.length ||
+            enabledEngines.some((id, i) => id !== prevEnabledEngines[i])))
+      ) {
+        // The regional filters engine is no longer used, so we must remove it
+        // from the storage. We do it as rarely as possible, to avoid unnecessary loads.
+        // TODO: this can be removed in the future release when most of the users will have
+        // the new version of the extension
+        engines.remove('regional-filters');
+
+        await reloadMainEngine();
+      }
 
-    // Experimental filters changed to enabled
-    if (lastValue !== undefined && value) {
-      await updateEngines();
-    }
-  }),
+      if (options.filtersUpdatedAt < Date.now() - HOUR_IN_MS) {
+        await updateEngines();
+      }
+    },
+  ),
+  OptionsObserver.addListener(
+    'experimentalFilters',
+    async (value, lastValue) => {
```

Comment:
add a function name

Reply:
It would be the same as the property name, and the function name is only used when there is no property. This is only for logging, not to make more understandable code.